File: src/academy/learnprogramming/interoperability/Nullability.java
package academy.learnprogramming.interoperability;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class Nullability {

    private String nonNullVariable;
    private String nullVariable;

    public Nullability(String nonNullVariable, String nullVariable) {
        this.nonNullVariable = nonNullVariable;
        this.nullVariable = nullVariable;
    }

    public void setNonNullVariable(@NotNull String nonNullVariable) {
        this.nonNullVariable = nonNullVariable;
    }

    public void setNullVariable(@Nullable String nullVariable) {
        this.nullVariable = nullVariable;
    }

    public @NotNull String getNonNullVariable() {
        return nonNullVariable;
    }

    public @Nullable String getNullVariable() {
        return nullVariable;
    }

    public String getNullVariable2() {
        return nullVariable;
    }
}

File: src/academy/learnprogramming/interoperability/Nullability.kt
package academy.learnprogramming.interoperability


fun main(args: Array<String>) {

    val sth = Nullability("not null", "null")

    //sth.setNonNullVariable(null)  // error

    //var sth2: String = sth.nullVariable  // error

    //platform type (!)
    //val sth3: Int = sth.nullVariable2  // error

}